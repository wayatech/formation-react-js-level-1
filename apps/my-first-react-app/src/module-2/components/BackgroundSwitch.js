import React from "react";

function BackgroundSwitch(props) {
  return (
    <React.Fragment>
      <button onClick={() => props.onClick("green")}>Green</button>
      <button onClick={() => props.onClick("red")}>Red</button>
    </React.Fragment>
  );
}

export default BackgroundSwitch;
