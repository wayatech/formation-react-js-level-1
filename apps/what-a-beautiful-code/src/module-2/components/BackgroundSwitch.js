import React from 'react'
import PropTypes from 'prop-types'

function BackgroundSwitch(props) {
  return (
    <>
      <button type="button" onClick={() => props.onClick('green')}>
        Green
      </button>
      <button type="button" onClick={() => props.onClick('red')}>
        Red
      </button>
    </>
  )
}

BackgroundSwitch.propTypes = {
  onClick: PropTypes.func.isRequired,
}

export default BackgroundSwitch
