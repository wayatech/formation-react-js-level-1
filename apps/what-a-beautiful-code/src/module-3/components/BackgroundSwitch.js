import React from 'react'
import PropTypes from 'prop-types'

const BackgroundSwitch = ({ onClick }) => (
  <>
    <button type="button" onClick={() => onClick('green')}>
      Green
    </button>
    <button type="button" onClick={() => onClick('red')}>
      Red
    </button>
  </>
)

BackgroundSwitch.propTypes = {
  onClick: PropTypes.func.isRequired,
}

export default BackgroundSwitch
