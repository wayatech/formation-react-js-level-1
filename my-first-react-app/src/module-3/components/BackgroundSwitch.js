import React from "react";

class BackgroundSwitch extends React.Component {
  constructor(props) {
    super(props);
    this.state = { seconds: 0 };
  }

  render() {
    return (
      <React.Fragment>
        <button onClick={() => this.props.onClick("green")}>Green</button>
        <button onClick={() => this.props.onClick("red")}>Red</button>
      </React.Fragment>
    );
  }
}

export default BackgroundSwitch;
