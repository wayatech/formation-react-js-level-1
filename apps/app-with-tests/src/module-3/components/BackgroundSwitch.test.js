import React from "react";
import renderer from "react-test-renderer";

import BackgroundSwitch from "./BackgroundSwitch";

test("BackgroundSwitch change color to green and red", () => {
  let value;
  const component = renderer.create(
    <BackgroundSwitch onClick={(val) => (value = val)} />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the click
  const buttons = component.root.findAllByType("button");
  buttons[0].props.onClick();
  expect(value).toBe("green");

  buttons[1].props.onClick();
  expect(value).toBe("red");
});
