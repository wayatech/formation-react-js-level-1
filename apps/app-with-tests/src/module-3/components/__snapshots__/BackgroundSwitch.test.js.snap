// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackgroundSwitch  correctly 1`] = `
Array [
  <button
    onClick={[Function]}
    type="button"
  >
    Green
  </button>,
  <button
    onClick={[Function]}
    type="button"
  >
    Red
  </button>,
]
`;

exports[`BackgroundSwitch change color to green and red 1`] = `
Array [
  <button
    onClick={[Function]}
    type="button"
  >
    Green
  </button>,
  <button
    onClick={[Function]}
    type="button"
  >
    Red
  </button>,
]
`;

exports[`BackgroundSwitch increment correctly 1`] = `
Array [
  <button
    onClick={[Function]}
    type="button"
  >
    Green
  </button>,
  <button
    onClick={[Function]}
    type="button"
  >
    Red
  </button>,
]
`;
